Tests/BSONTests/BSONTests.swift

```diff
@@ -1 +1,143 @@
 import BSON
+import Foundation
+import Nimble
+import NIO
+import XCTest
+
+open class BSONTestCase: XCTestCase {
+    /// Gets the path of the directory containing spec files, depending on whether
+    /// we're running from XCode or the command line
+    static var specsPath: String {
+        // if we can access the "/Tests" directory, assume we're running from command line
+        if FileManager.default.fileExists(atPath: "./Tests") {
+            return "./Tests/Specs"
+        }
+        // otherwise we're in Xcode, get the bundle's resource path
+        guard let path = Bundle(for: self).resourcePath else {
+            XCTFail("Missing resource path")
+            return ""
+        }
+        return path
+    }
+
+    struct TestError: Error {
+        let message: String
+    }
+
+    func fail(_ message: String = "This should not happen") -> Never {
```

Comment:
are we using this anywhere? / if you don't want tests to keep running after failures are encountered the preferred way is to set `continueAfterFailure` to false, like here: https://github.com/mongodb/mongo-swift-driver/blob/26b17296a96ab6a0255bbbf1a86e71565ce3f3c0/Tests/MongoSwiftSyncTests/RetryableReadsTests.swift#L53